internal/cli/pipeline_get_run.go

```diff
@@ -0,0 +1,232 @@
+			pipelineIdent, app.Ref().Application, terminal.WithHeaderStyle())
+
+		sg := app.UI.StepGroup()
+		defer sg.Wait()
+
+		step := sg.Add("Reading pipeline run...")
+		defer step.Abort()
+
+		// build the base api request
+		getPipelineReq := &pb.GetPipelineRequest{}
+
+		// Reference by ID if set, otherwise by name
+		if c.flagPipelineId != "" {
+			getPipelineReq.Pipeline = &pb.Ref_Pipeline{
+				Ref: &pb.Ref_Pipeline_Id{
+					Id: c.flagPipelineId,
+				},
+			}
+		} else {
+			getPipelineReq.Pipeline = &pb.Ref_Pipeline{
+				Ref: &pb.Ref_Pipeline_Owner{
+					Owner: &pb.Ref_PipelineOwner{
+						Project:      &pb.Ref_Project{Project: app.Ref().Project},
+						PipelineName: pipelineName,
+					},
+				},
+			}
+		}
+
+		step.Update("Requesting to stream run of pipeline %q...", pipelineIdent)
+
+		var (
+			resp *pb.GetPipelineRunResponse
+			err  error
+		)
+		if c.flagRunSequence == 0 {
+			// take pipeline id and queue a RunPipeline with a Job Template.
+			resp, err = c.project.Client().GetLatestPipelineRun(c.Ctx, getPipelineReq)
+			if err != nil {
+				return err
+			}
+		} else {
+			// take pipeline id and queue a RunPipeline with a Job Template.
+			resp, err = c.project.Client().GetPipelineRun(c.Ctx, &pb.GetPipelineRunRequest{
+				Pipeline: getPipelineReq.Pipeline,
+				Sequence: uint64(c.flagRunSequence),
+			})
+			if err != nil {
+				return err
+			}
+		}
+		if resp == nil {
+			app.UI.Output("Getting a pipeline run returned a nil response", terminal.WithErrorStyle())
+			return fmt.Errorf("Response was empty when requesting a pipeline run for pipeline %q", pipelineIdent)
+		}
+
+		step.Update("Attempting to read job stream sequentially in order")
+		step.Done()
+
+		// Receive job ids from running pipeline, use job client to attach to job stream
+		// and stream here. First pass can be linear job streaming
+		step = sg.Add("")
+		defer step.Abort()
+
+		steps := len(resp.PipelineRun.Jobs)
+		step.Update("%d steps detected, run sequence %d, pipeline run status %q",
+			steps, resp.PipelineRun.Sequence, pb.PipelineRun_State_name[int32(resp.PipelineRun.State)])
+		step.Done()
+
+		successful := steps
+		for _, jobIdRef := range resp.PipelineRun.Jobs {
+			jobId := jobIdRef.Id
+			job, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
+				JobId: jobId,
+			})
+			if err != nil {
+				return err
+			}
+			// NOTE(briancain): We intentionally skip Noop type jobs because currently
+			// we make step Refs for pipelines run a Noop job to make dependency tracking
+			// for pipeline step refs easier. We don't stream a noop output job because
+			// there's nothing to stream.
+			if _, ok := job.Operation.(*pb.Job_Noop_); ok {
+				continue
+			}
+
+			ws := "default"
+			if job.Workspace != nil {
+				ws = job.Workspace.Workspace
+			}
+			// TODO(briancain): fix me
+			//app.UI.Output("Executing Step %q in workspace: %q", resp.JobMap[jobId].Step, ws, terminal.WithHeaderStyle())
```

Comment:
We might want to handle this in a follow-up PR prior to the next release, but ideally if we want to display step names next to job streams like we do for `waypoint pipeline run` we'll need to modify the Run API endpoints to calculate the JobMap in the same way `RunPipeline` does.

Reply:
Oh wait this is easier than I thought. We store this directly on the job, so we can grab it there...